multiagenticswarm/core/prompt_parser.py

```diff
@@ -0,0 +1,204 @@
+# multiagenticswarm/core/prompt_parser.py
+import re
+import json
+import os
+from typing import Dict, Any, Optional, Callable
+from langchain_core.messages import SystemMessage, HumanMessage
+from langchain_groq import ChatGroq
+from dotenv import load_dotenv
+
+# Load environment variables
+load_dotenv()
+
+# Import schema + role mappings
+from prompt_schemas import DEFAULT_FUNCTION_SCHEMA, ROLE_MAPPINGS
+print(DEFAULT_FUNCTION_SCHEMA)
+print(ROLE_MAPPINGS)
+
+# FUNCTION CALL PROMPT BUILDER
+def build_enhanced_function_call_prompt(prompt: str):
+    system_content = f"""You are an expert workflow parser for multi-agent collaboration systems.
```

Comment:
System prompt embedded in code makes it inflexible. Never hardcode anything in prod code

Fix: Load system promt from configuration:

something like this:
from ..config import GraphCompilationConfig

def build_prompt(prompt: str, config: GraphCompilationConfig):
    return [
        SystemMessage(content=config.parser_system_prompt),
    ]